### PR TITLE
TM-1785: More fields added to BidList export. TM-1595: cdo client BidList export endpoint added.

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
[TM-1785](https://metaphase.atlassian.net/browse/TM-1785)
[TM-1595](https://metaphase.atlassian.net/browse/TM-1595)

### CODE REVIEW

File: talentmap_api/fsbid/services/available_positions.py [hunk 1]
```diff
@@ -47,6 +46,18 @@ def get_unavailable_position(id, jwt_token):
...
 
+def get_all_position(id, jwt_token):
+    '''
+    Gets an indivdual position by id
+    '''
+
+    return services.get_individual(
+        "availablePositions",
+        id,
+        convert_ap_query,
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:57]: @mjoyce91 Per our discussion: If I put `convert_all_query` here, it breaks, even breaks if I put `convert_up_query` it breaks.

File: talentmap_api/fsbid/services/available_positions.py [hunk 2]
```diff
@@ -268,3 +279,6 @@ def convert_ap_query(query):
 
 def convert_up_query(query):
     return (convert_ap_query(query, "FP"))
+
+def convert_all_query(query):
+    return (convert_ap_query(query, "FP,OP,HS"))
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:284]: Also, `convert_ap_query` doesn't even take a second parameter 🤔 so I'm unsure of the functionality of `convert_up_query`?

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 @scott062 

Do Bids get removed from bidlist when the season is over?

File: talentmap_api/fsbid/services/available_positions.py (hunk 2 above)
Comment (mjoyce91) [reply talentmap_api/fsbid/services/available_positions.py:284]: I think that might have changed when we updated OP/HS search a little awhile ago

File: talentmap_api/fsbid/services/available_positions.py (hunk 1 above)
Comment (mjoyce91) [reply talentmap_api/fsbid/services/available_positions.py:57]: What's the error?

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/available_positions.py:57]: The outputs in the terminal just stop lol and I wait for a while too. I can out more print statements and see where it breaks in `convert_up_query`

File: talentmap_api/fsbid/services/available_positions.py (hunk 2 above)
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/available_positions.py:284]: ah, gotcha. So we might not even need `convert_up_query`?

Comment (mjoyce91) [reply talentmap_api/fsbid/services/available_positions.py:284]: I'll need to check

**Comment by mjoyce91** [comment]
> Do Bids get removed from bidlist when the Bidding Season is over?

We don't really handle that on our side, but more or less, yes

**Comment by elizabeth-jimenez** [comment]
> > Do Bids get removed from bidlist when the Bidding Season is over?
> 
> We don't really handle that on our side, but more or less, yes

Ah gotcha, thanks! I ask because if they do and `data` on line 37 of file `talentmap_api/fsbid/services/bid.py` only returns available positions, then we may be able to just use `get_available_position` line 25 of file `talentmap_api/fsbid/services/available_positions.py`, right?

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -338,46 +338,47 @@ def get_bids_csv(data, filename):
...
-        # row.append(ted)
-        # row.append(smart_str(record["position"]["current_assignment"]["user"]))
-        row.append(smart_str(record["bidcycle"]))
-        row.append(smart_str(record.get("status")))
-        # row.append(smart_str(record["position"]["description"]["content"]))
-
-        writer.writerow(row)
+        position_data = get_all_position(smart_str(record["position"]["id"]), jwt_token)
+        if position_data is not None:
+            # Mike: my concern here was that I think that the data coming through for bids should only be
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:357]: 👨‍💻

**Comment by mjoyce91** [comment]
>just use get_available_position line 25 of file talentmap_api/fsbid/services/available_positions.py, instead of get_all_position

No, you'll still need the unavailable position endpoint. HS is available, but once the bid is approved, it becomes unavailable. It's still possible to see unavailable positions in your current bid list.

**Comment by elizabeth-jimenez** [comment]
> > just use get_available_position line 25 of file talentmap_api/fsbid/services/available_positions.py, instead of get_all_position
> 
> No, you'll still need the unavailable position endpoint. HS is available, but once the bid is approved, it becomes unavailable. It's still possible to see unavailable positions in your current bid list.

Ahhhh, I see. That explains it. Thanks Mike!

**Comment by mjoyce91** [comment]
@elizabeth-jimenez - is this ready to review with your latest commits?

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez - is this ready to review with your latest commits?

Yes! 👩‍💻

File: talentmap_api/fsbid/services/common.py [hunk 4]
```diff
@@ -338,46 +338,45 @@ def get_bids_csv(data, filename):
...
+                ted = smart_str(maya.parse(position_data["ted"]).datetime().strftime('%m/%d/%Y'))
+            except:
+                ted = "None listed"
+
+            row = []
+            row.append(smart_str(record["position"]["title"]))
+            row.append(smart_str("=\"%s\"" % record["position"]["position_number"]))
+            row.append(smart_str(record["position"]["skill"]))
+            row.append(smart_str("=\"%s\"" % record["position"]["grade"]))
+            row.append(smart_str(record["position"]["bureau"]))
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/common.py:367]: Is bureau coming through in the CSV for you?

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/common.py:367]: 😱 no!

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/common.py]: updated 😅

**Pull request merged**